condacolab.py

```diff
@@ -87,12 +95,114 @@ def _run_subprocess(command, logs_filename):
+    channels
+        Comma separated list of channels to use in the conda
+        base environment.
+    pip_args
+        List of additional packages to be installed using pip.
+    extra_conda_args
+        Any extra conda arguments to be used during the installation.
+    """
+    os.makedirs("/var/condacolab", exist_ok=True)
+    environment_file_path = "/var/condacolab/environment.yaml"
+
+    # When environment.yaml file is not provided.
+    if environment_file is None:
+        env_details = {}
+        if channels:
+            env_details["channels"] = channels
+        if specs:
+            env_details["dependencies"] = specs
+        if python_version:
+            env_details["dependencies"] += [f"python={python_version}"]
+        if pip_args:
+            pip_args_dict = {"pip": pip_args}
+            env_details["dependencies"].append(pip_args_dict)
+
+        with open(environment_file_path, 'w') as f:
+            yaml.indent(mapping=2, sequence=4, offset=2)
+            yaml.dump(env_details, f)
+    else:
+        # If URL is given for environment.yaml file
+        if environment_file.startswith(("http://", "https://")):
+            try:
+                with urlopen(environment_file) as response, open(environment_file_path, "wb") as out:
+                    shutil.copyfileobj(response, out)
+            except HTTPError as e:
+                raise HTTPError("The URL you entered is not working, please check it again.") from e
+
+        # If path is given for environment.yaml file
+        else:
+            shutil.copy(environment_file, environment_file_path)
+
+        with open(environment_file_path, 'r') as f:
+            env_details = yaml.load(f.read())
+
+        for key in env_details:
+            if channels and key == "channels":
+                env_details["channels"].extend(channels)
+            if key == "dependencies":
+                if specs:
+                    env_details["dependencies"].extend(specs)
+                if python_version:
+                    env_details["dependencies"].extend([f"python={python_version}"])
+                if pip_args:
+                    for element in env_details["dependencies"]:
+                        # if pip dependencies are already specified.
+                        if isinstance(element, CommentedMap) and "pip" in element:
+                            element["pip"].extend(pip_args)
+                            break
+                        # if there are no pip dependencies specified in the yaml file.
+                    else:
+                        pip_args_dict = CommentedMap([("pip", [*pip_args])])
+                        env_details["dependencies"].append(pip_args_dict)
+
+        with open(environment_file_path, 'w') as f:
+            f.truncate(0)
+            yaml.dump(env_details, f)
+
+    _run_subprocess(
+        [f"{prefix}/bin/python", "-m", "conda_env", "update", "-n", "base", "-f", environment_file_path, *extra_conda_args],
```

Comment:
Let's pass `conda_exe` from the calling function to this one so we can use `mamba` if available. That should reduce solver times.

```suggestion
        [conda_exe, "env", "update", "-n", "base", "-f", environment_file_path, *extra_conda_args],
```